test/cri-containerd/removepodsandbox_test.go

```diff
@@ -0,0 +1,167 @@
+// +build functional
+
+package cri_containerd
+
+import (
+	"context"
+	"testing"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	runtime "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+)
+
+type TestConfig struct {
+	name             string
+	containerName    string
+	requiredFeatures []string
+	runtimeHandler   string
+	sandboxImage     string
+	containerImage   string
+	cmd              []string
+}
+
+// Utility function to test removal of a sandbox with no containers and no previous call to stop the pod
+func runPodSandboxTestWithoutPodStop(t *testing.T, request *runtime.RunPodSandboxRequest) {
+	client := newTestRuntimeClient(t)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	podId := runPodSandbox(t, client, ctx, request)
+	removePodSandbox(t, client, ctx, podId)
+
+	_, err := client.PodSandboxStatus(ctx, &runtime.PodSandboxStatusRequest{
+		PodSandboxId: podId,
+	})
+
+	status, ok := status.FromError(err)
+	if !ok || status.Code() != codes.NotFound {
+		t.Fatalf("PodSandboxStatus did not return expected errorStatus: %s", err)
+	}
+}
+
+// Utility function to start sandbox with one container and make sure that sandbox is removed in the end
+func runPodWithContainer(t *testing.T, client runtime.RuntimeServiceClient, ctx context.Context, tc *TestConfig) (string, string) {
+	request := getRunPodSandboxRequest(t, tc.runtimeHandler)
+	podId := runPodSandbox(t, client, ctx, &request)
+	defer removePodSandbox(t, client, ctx, podId)
+
+	annotations := map[string]string{}
+	mounts := []*runtime.Mount{}
```

Comment:
Do you know if these are needed or if we can just pass through nil in `createContainerInSandbox` for those fields?

Reply:
of course we can just pass `nil` 😄